Lib/test/libregrtest/setup.py

```diff
@@ -104,10 +105,10 @@ def _test_audit_hook(name, args):
 
     # Ensure there's a non-ASCII character in env vars at all times to force
     # tests consider this case. See BPO-44647 for details.
-    os.environ.setdefault(
-        UNICODE_GUARD_ENV,
-        "\N{SMILING FACE WITH SUNGLASSES}",
-    )
+    if TESTFN_UNDECODABLE and hasattr(os, 'environb'):
+        os.environb.setdefault(UNICODE_GUARD_ENV.encode(), TESTFN_UNDECODABLE)
```

Comment:
os.environb is a wrapper to os.environ. You can use os.fsencode()/os.fsdecode() to set the key in os.environ.

Reply:
Rather `os.environ` is a wrapper to `os.environb`.

Reply:
It's fine if you want to use os.environb. It's just that I regret that I added it: surrogateescape of PEP 383 is a better solution to use Unicode (os.environ) on all platforms. It avoids treating Unix differently. os.environb is just here to reduce suprises for Unix users. Internally, os.environ and os.environb are sharing the same Python dict.

Reply:
And this dict contains bytes.

If use `os.environ`, there will be double conversion: `encode(decode(value))`. I am not sure that it is lossless in all cases.